sdk/src/main/java/io/customer/sdk/device/DeviceTokenProvider.kt

```diff
@@ -0,0 +1,11 @@
+package io.customer.sdk.device
+
+import android.content.Context
+
+/**
+ *  Responsible for token generation and validity
+ */
+interface DeviceTokenProvider {
```

Comment:
Moved it out from the FCM package, because we can have more integrations e.g OneSignal etc and this should be available to all modules.

Reply:
The iOS SDK has a base `messaging-push` module that stores common code shared between the `messaging-push-fcm` and `messaging-push-apn` modules. 

Should we do the same for this code instead of putting it in the tracking module? Since this interface is only needed by `messaging-push` modules?

Reply:
Good point, but not sure what benefit would it provide since `the iOS` module does have methods that are shared among the other two modules, `messaging-push-fcm` and `messaging-push-apn`.

But in Android, `messaging-push-fcm` just utilizes the `CustomerIO` public methods. So having a separate module, just for one class seems like an over-kill. 

In the future when we do see ourselves finding a use-case where we have more common code, we can create a base `messaging-push` module and move these classes over there?